src/components/form/ha-form.ts

```diff
@@ -0,0 +1,179 @@
+  @property() public computeSuffix?: (schema: HaFormSchema) => string;
+  @query("ha-form") private _childForm?: HaForm;
+  @query("#element") private _elementContainer?: HTMLDivElement;
+
+  public focus() {
+    const input = this._childForm
+      ? this._childForm
+      : this._elementContainer
+      ? this._elementContainer.lastChild
+      : undefined;
+
+    if (!input) {
+      return;
+    }
+
+    (input as HTMLElement).focus();
+  }
+
+  protected render() {
+    if (Array.isArray(this.schema)) {
+      return html`
+        ${this.error && this.error.base
+          ? html`
+              <div class="error">
+                ${this._computeError(this.error.base, this.schema)}
+              </div>
+            `
+          : ""}
+        ${this.schema.map(
+          (item) => html`
+            <ha-form
+              .data=${this._getValue(this.data, item)}
+              .schema=${item}
+              .error=${this._getValue(this.error, item)}
+              @value-changed=${this._valueChanged}
+              .computeError=${this.computeError}
+              .computeLabel=${this.computeLabel}
+              .computeSuffix=${this.computeSuffix}
+            ></ha-form>
+          `
+        )}
+      `;
+    }
+
+    return html`
+      ${this.error
+        ? html`
+            <div class="error">
+              ${this._computeError(this.error, this.schema)}
+            </div>
+          `
+        : ""}
+      <div id="element" @value-changed=${this._valueChanged}></div>
+    `;
+  }
+
+  protected updated(changedProperties: PropertyValues) {
+    const oldSchema: HaFormSchema = changedProperties.get(
+      "schema"
+    ) as HaFormSchema;
+    if (
+      !Array.isArray(this.schema) &&
+      (!oldSchema || oldSchema.type !== this.schema.type)
+    ) {
+      const element = document.createElement(
+        `ha-form-${this.schema.type}`
+      ) as HaFormElement;
+      element.schema = this.schema;
+      element.data = this.data;
+      element.label = this._computeLabel(this.schema);
+      element.suffix = this._computeSuffix(this.schema);
+      if (this._elementContainer!.lastChild) {
+        this._elementContainer!.removeChild(this._elementContainer!.lastChild);
+      }
+      this._elementContainer!.append(element);
+    } else if (this._elementContainer && this._elementContainer.lastChild) {
+      const element = this._elementContainer!.lastChild as HaFormElement;
+      element.schema = this.schema;
+      element.data = this.data;
+      element.label = this._computeLabel(this.schema);
+      element.suffix = this._computeSuffix(this.schema);
+    }
+  }
+
+  private _computeLabel(schema) {
+    return this.computeLabel
+      ? this.computeLabel(schema)
+      : schema
+      ? schema.name
+      : "";
+  }
+
+  private _computeSuffix(schema) {
+    return this.computeSuffix
+      ? this.computeSuffix(schema)
+      : schema && schema.description
+      ? schema.description.unit_of_measurement
```

Comment:
I wonder if this should have been `schema.description.suffix` and that we made a mistake referring to it as `unit_of_measurement`. It doesn't scale to other things.